File: engine/src/com/magma/engine/gfx/AnimSlot.java
package com.magma.engine.gfx;

// A convoluted way to not compare strings
public class AnimSlot {
    private static int nextIndex;
    
    public final int index;
    public final float fps;
    public final String name;
    
    public AnimSlot(String name, float fps){
        this.index = nextIndex;
        this.fps = fps;
        this.name = name;
        nextIndex++;
    }
    
    public static AnimSlot of(String name, float fps) {
    	return new AnimSlot(name,fps);
    }
    public static AnimSlot of(String name) {
    	return new AnimSlot(name,2);
    }
}

File: engine/src/com/magma/engine/gfx/AnimSlot.kt
package com.magma.engine.gfx

// A convoluted way of working that uses indices instead of comparing strings = more performance
class AnimSlot(val name: String, val fps: Float) {
    val index: Int = nextIndex

    init {
        nextIndex++
    }

    companion object {
        private var nextIndex = 0
        fun of(name: String, fps: Float): AnimSlot {
            return AnimSlot(name, fps)
        }
        fun of(name: String): AnimSlot {
            return AnimSlot(name, 2f)
        }
    }
}